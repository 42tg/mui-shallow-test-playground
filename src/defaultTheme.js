import { green, blue } from "@material-ui/core/colors";

export default {
  palette: {
    primary: blue,
    secondary: green
  },
  dimensions: {
    font: "2em"
  }
};
