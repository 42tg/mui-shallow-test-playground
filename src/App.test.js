import React, { Provider } from "react";
import Button from "./components/Buttons";
import { ThemeProvider, createMuiTheme } from "@material-ui/core";
import defaultTheme from "./defaultTheme";

import { createShallow, createMount } from "@material-ui/core/test-utils";

describe("<Button />", () => {
  let shallow;
  beforeAll(() => {
    shallow = createShallow({
      wrappingComponent: ({ children, theme }) => (
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      ),
      wrappingComponentProps: {
        theme: createMuiTheme(defaultTheme)
      }
    });
  });
  it("Should render the button", () => {
    const wrapper = shallow(<Button>Hello World</Button>);
    expect(wrapper.container).toMatchSnapshot();
  });
});

describe("<Button />", () => {
  let mount;
  beforeAll(() => {
    mount = createMount({
      wrappingComponent: ({ children, theme }) => (
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      ),
      wrappingComponentProps: {
        theme: createMuiTheme(defaultTheme)
      }
    });
  });
  it("Should render the button", () => {
    const wrapper = mount(<Button>Hello World</Button>);
    expect(wrapper.container).toMatchSnapshot();
  });
});
