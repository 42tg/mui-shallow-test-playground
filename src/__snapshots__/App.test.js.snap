// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should render the button 1`] = `undefined`;
